proxy/src/config.rs

```diff
@@ -517,6 +519,59 @@ impl FromStr for ProjectInfoCacheOptions {
     }
 }
 
+/// This is a config for connect to compute and wake compute.
+#[derive(Clone, Copy, Debug)]
+pub struct RetryConfig {
+    /// Number of times we should retry.
+    pub max_retries: u32,
+    /// Retry duration is base_delay * backoff_factor ^ n, where n starts at 0
+    pub base_delay: tokio::time::Duration,
+    /// Exponential base for retry wait duration
+    pub backoff_factor: f64,
+}
+
+impl RetryConfig {
+    /// Default options for RetryConfig.
+
+    /// Total delay for 4 retries with 1s base delay and 2.0 backoff factor is 7s.
+    pub const CONNECT_TO_COMPUTE_DEFAULT_VALUES: &'static str =
+        "num_retries=4,base_retry_wait_duration=1s,retry_wait_exponent_base=2.0";
+    /// Total delay for 4 retries with 1s base delay and 2.0 backoff factor is 7s.
+    /// Cplane has timeout of 60s on each request.
+    pub const WAKE_COMPUTE_DEFAULT_VALUES: &'static str =
+        "num_retries=4,base_retry_wait_duration=1s,retry_wait_exponent_base=2.0";
```

Comment:
hmm. I don't think I noticed these defaults the first time. @khanova I feel like 1 second wait time for the first retry is too long. I've looked in prod and some cases connect after 1.4s. We would fail on 1s and then succeed at 3s, doubling our effective latency.

A number picked at random, how about:
`"num_retries=8,base_retry_wait_duration=100ms,retry_wait_exponent_base=1.6"`
The total time is `(1-1.6^8) / (1-1.6) * 100ms = 6.99s`

Reply:
Ok. Should we do it for connect to compute or to wake compute as well?

Reply:
both I think. I don't recall us hitting wake compute retries much. normally happens when an operation is in progress and _hopefully_ the operations are quick. our new endpoint wake_compute limiter should kick in anyway

Reply:
https://github.com/neondatabase/neon/pull/7509